pkg/detectors/atlassian/v2/atlassian.go

```diff
@@ -58,31 +59,52 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		uniqueMatches[match[1]] = struct{}{}
 	}
 
-	for match := range uniqueMatches {
-		s1 := detectors.Result{
-			DetectorType: detectorspb.DetectorType_Atlassian,
-			Raw:          []byte(match),
-			ExtraData: map[string]string{
-				"rotation_guide": "https://howtorotate.com/docs/tutorials/atlassian/",
-				"version":        fmt.Sprintf("%d", s.Version()),
-			},
-		}
+	uniqueOrgIdMatches := make(map[string]struct{})
+	for _, match := range organizationIdPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueOrgIdMatches[match[1]] = struct{}{}
+	}
+	if len(uniqueOrgIdMatches) == 0 {
+		// we only need an org ID to pass into AnalysisInfo
+		// if we don't find one, we can still verify the key
+		// we can add a dummy entry here just to make sure a result is returned
+		uniqueOrgIdMatches[""] = struct{}{}
+	}
 
-		if verify {
-			client := s.client
-			if client == nil {
-				client = defaultClient
+	for match := range uniqueMatches {
+		for orgId := range uniqueOrgIdMatches {
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_Atlassian,
+				Raw:          []byte(match),
+				ExtraData: map[string]string{
+					"rotation_guide": "https://howtorotate.com/docs/tutorials/atlassian/",
+					"version":        fmt.Sprintf("%d", s.Version()),
+				},
 			}
 
-			isVerified, orgResponse, verificationErr := verifyMatch(ctx, client, match)
-			s1.Verified = isVerified
-			if orgResponse != nil && len(orgResponse.Data) > 0 {
-				s1.ExtraData["Organization"] = orgResponse.Data[0].Attributes.Name
+			if verify {
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+
+				isVerified, orgResponse, verificationErr := verifyMatch(ctx, client, match)
+				s1.Verified = isVerified
+				if orgResponse != nil && len(orgResponse.Data) > 0 {
+					s1.ExtraData["Organization"] = orgResponse.Data[0].Attributes.Name
+				}
+				s1.SetVerificationError(verificationErr, match)
+				if s1.Verified {
+					s1.AnalysisInfo = map[string]string{
+						"key": match,
+					}
+					if orgId != "" {
+						s1.AnalysisInfo["organization_id"] = orgId
+					}
```

Comment:
Is it possible that some secrets may not actually belong to the `orgId` we’re sending here? From what I see, we’re always passing the `orgId` regardless of whether the secret is tied to that organization. How can we be sure that this `orgId` will work with this secret in the analyzer?
Does that approach make sense in this context?

Reply:
It may be possible that the secret might not belong to the `orgId`, and in that case it won't work with the secret in the analyzer. But that's only with Scoped API keys though, for Unscoped (Full Access) keys, we do not use the `organization_id` in the analyzer, irrespective of it being provided or not.

For scoped keys, passing an incorrect `organization_id` is the same as not passing an `organization_id`, as it will error out in both cases. So there's really no benefit in validating that the `organization_id` belongs to the secret.